src/window.cpp

```diff
@@ -775,15 +774,16 @@ void WindowManager::AddToStack(Window *w)
 /**
  * Remove a window from the list.
  * @param w Window to remove.
+ * @param about_to_be_deleted The window will be deleted soon after.
  */
-void WindowManager::RemoveFromStack(Window *w)
+void WindowManager::RemoveFromStack(Window *w, bool about_to_be_deleted)
```

Comment:
Not sure about the flag here. The name seems to imply the window is deleted somewhat, doesn't it?

Wrong use could lead to crashes due to the `current_window` pointing to a deleted window.

Perhaps there should be several functions here, one for each use?
Then you can move this flag inside the window manager code without exposing it publically. Alternatively, this function could crash if you try to remove a current window (making it the responsibility of the caller to handle it)